Fix the extension 'post_step' call always being passed 'failed' status

ehuelsmann opened the pull request:
Since $result is a T::B::C::Model::Result, fix the value being passed
to the extension by passing the actual test result value.

pjlsergeant commented:
While I'm not convinced that extensions are being widely used yet, it seems your change here would break the API for them. How about adding the result itself as a fourth argument, rather than changing the type of the third argument?

ehuelsmann commented:
Good point. I'll fix the third argument (which is currently always `false`) and move the full step result to the fourth parameter -- then update the PR.

ehuelsmann commented:
Updated this PR to leave in place the original argument (but fixed to no longer always be 'false') and added an additional argument for further result access.
